### ISSUE TITLE
Configuring Separate Intra-node and Inter-node Communication in 2D Topology with Scale Up and Scale Out Networks in Astra-Sim1.0 + NS3

### ISSUE THREAD

**Issue by gitzhangzhao** [astra-sim/astra-sim#273]
For the All to All communication in a 2D logical topology, I have defined Scale up (e.g., NVLink between 8 GPUs within a node) and Scale out networks (e.g., Ethernet between nodes) based on real-world scenarios. Scale up has higher bandwidth and lower latency. 

However, when I trace each GPU node, I notice that a portion of the All to All traffic is going through Scale up, while another portion is going through Scale out. I want to ensure that intra-node communication occurs over the Scale up network, while inter-node communication occurs over the Scale out network. 

How can I configure this? 

I am using Astra-Sim 1.0 + NS3.

**Comment by Tejanikhil-MSR** [comment]
Hi, 

Wanted to ask if you are able to simulate the configuration that you have asked before.

**Comment by gitzhangzhao** [comment]
> Hi, 
> 
> 
> 
> Wanted to ask if you are able to simulate the configuration that you have asked before. 
> 
> 
> 
> > For the All to All communication in a 2D logical topology, I have defined Scale up (e.g., NVLink between 8 GPUs within a node) and Scale out networks (e.g., Ethernet between nodes) based on real-world scenarios. Scale up has higher bandwidth and lower latency.

I found that ns-3 determines the traffic distribution based on the number of hops between nodes. If the number of hops between two domains is the same, the traffic will be evenly distributed. Therefore, in order to simulate the network topology in real situations, I added an extra layer of switches in the scale-out architecture. In this way, the communication between GPUs within the same machine will all go through the scale-out architecture, and the communication between different machines will also go through the scale-out architecture. However, this is only a temporary solution. Due to the tight schedule of my project, I had no choice but to adopt this approach. In fact, there should be better methods.

**Comment by Tejanikhil-MSR** [comment]
> Therefore, in order to simulate the network topology in real situations, I added an extra layer of switches in the scale-out architecture

You mean to say that you have modified the existing code to support real-time network topologies.

**Comment by gitzhangzhao** [comment]
> > Therefore, in order to simulate the network topology in real situations, I added an extra layer of switches in the scale-out architecture
> 
> You mean to say that you have modified the existing code to support real-time network topologies.

No, I haven't modified the relevant code. I just made changes to the topology configuration file.
